### PR TITLE
Created example 6a, 6b and 6c

### PR DESCRIPTION
**Pull request by hrishiv**
Created examples related to contact. Ref #225

### CODE REVIEW

File: doc/config.yml [hunk 1]
```diff
@@ -33,6 +33,14 @@ Extensions:
...
                 Example 3: examples/example3.md
                 Example 4: examples/example4.md
+<<<<<<< HEAD
+=======
+                Example 5a: examples/example5a.md
+                Example 5b: examples/example5b.md
+                Example 6a: examples/example6a.md
+                Example 6b: examples/example6b.md
+                Example 6c: examples/example6c.md
+>>>>>>> 5dc8281... Created example 6a, 6b and 6c
```
Comment (hoffwm) [inline doc/config.yml]: Delete "<<<<<< HEAD" stuff

File: doc/content/examples/example6a.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Example 6a: Displacement Controlled Contact Problem with elastic soil
+
+## Model Description
+
+This example demonstrates the use of the contact block to simulate a frictional contact between a solid block and an elastic soil in MASTODON. A solid cubical block of dimensions 6" each side  is placed in contact with an cubical soil block of dimensions 48" each side. The remaining three sides of the soil are fixed and the solid block is displaced by a prescribed distance in the direction parallel to the contact surface. The normal forces and the frictional forces obtained from MASTODON is compared with the analytical results.
```
Comment (hoffwm) [inline doc/content/examples/example6a.md]: This example demonstrates frictional contact between a solid block and elastic soil in MASTODON. A 6 inch cubic block is centered on top of a 4 foot cubic block of soil. The top surface of the soil is free and the remaining surfaces are fixed in all directions. A uniform normal pressure of 5 psi is applied to the top of the smaller solid block and it is given a prescribed displacement in a direction parallel to the contact surface. The resulting normal and frictional forces at the interface of the two materials are then obtained and compared with analytical results.

File: doc/content/examples/example6a.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+
+## Modeling in MASTODON
+
+!listing examples/ex06a/psf_grav.i
+
+The mesh consisting of the solid block and the soil is generated by importing an existing mesh file [fig:ex06input].
+The details about the type of mesh file supported can be found at [GettingStarted](manuals/user/index.md). The displacements are defined in the variables section and the velocity, acceleration, normal forces and the tangential (frictional) forces are defined as the auxiliary variables.
```
Comment (hoffwm) [inline doc/content/examples/example6a.md]: The mesh used in this model was generated externally using the mesh generation software, Cubit. MASTODON imports the mesh using the 'FileMesh' type specified in the 'Mesh' block. Details regarding supported mesh file types can be found here...

The displacement variables are defined in the 'Variables' block. The accelerations, velocities, normal and tangential (frictional) forces are defined as auxiliary variables in the 'AuxVariables' block.

File: doc/content/examples/example6a.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+!media media/examples/ex06/ex06_input.png
+       style=width:50%;margin-left:150px;float:center;
+       id=fig:ex06input
+       caption=Input model in MASTODON.
+
+
+[TensorMechanics](syntax/index.md) and [InertialForce](syntax/index.md) kernels are used to model the dynamics of the 3D solid elements. [Gravity](syntax/index.md) kernels is used to apply the gravitational force. The [Newmark time integration](manuals/theory/index.md) parameters used in this problem correspond to the Newmark's average acceleration method, i.e. `beta = 0.25` and `gamma = 0.5` and damping is ignored.
+
+The [ComputeIsotropicElasticityTensorBeam](syntax/index.md) block is used to create the elasticity tensor of the solid block and the elastic soil, using the parameters, Young's Modulus and Poisson's ratio. The stresses and strain is calculated using [ComputeFiniteStrainElasticStress](syntax/index.md) and [ComputeFiniteStrain](syntax/index.md). The densities are assigned to the solid block and the soil using [GenericConstantMaterial](syntax/index.md).
```
Comment (hoffwm) [inline doc/content/examples/example6a.md]: ...block is used to create the elasticity tensor of the solid block and the elastic soil using Young's Modulus and Poisson's ratio. The stresses and strains are calculated using

File: doc/content/examples/example6a.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+The bottom and the sides of the soil are fixed. A uniform vertical pressure of 5 psi is applied at the top of the solid block and the block is displaced by the prescribed amount as shown in [input_motion].
```
Comment (hoffwm) [inline doc/content/examples/example6a.md]: In the soil block, the top surface is free and all other surfaces are fixed in every direction. A uniform vertical pressure of 5 psi is applied to the top surface of the solid block and it is given the prescribed displacement history shown in

File: doc/content/examples/example6a.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+!plot scatter filename=examples/ex06a/input_motion.csv
+              data=[{'x':'time', 'y':'displacement'}]
+              layout={'xaxis':{'title':'Time (secs)'},'yaxis':{'title':'Displacement (in)'}, 'title':'Displacement of the block'}
+              caption=Input displacement of the block
+              id=input_motion
+
+[Contact](manuals/user/index.md) is implemented using the Contact block. The slave and the master surfaces are defined in the contact block to implement slave-master concept for the contact problem which prevents the nodes of the slave boundary to penetrate the surface elements of the master boundary. Columb friction model with a friction coefficient of 0.7 is used. The penalty method with a penalty parameter of 1e5 is used to solve the contact problem. The penalty method allows small penetration and the contact force is proportional to it. Large penalty allows small penetration.
```
Comment (hoffwm) [inline doc/content/examples/example6a.md]: ...is defined in MASTODON using the 'Contact' block. This model is simulating Coulomb friction, with a coefficient of friction value equal to 0.7. The formulation implemented and demonstrated in this example uses a master/slave concept in which the nodes of the slave boundary are restricted by the nodes of the master surface boundary. Penetration is enforced using a penalty factor which is inversely related to the amount of penetration allowed (large penalty factor results in small amount of penetration and vice versa). The factor used in this example is 1e5.

File: doc/content/examples/example6a.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+!listing examples/ex06a/psf_grav.i start=Contact end=Preconditioning
+
+The control blocks is provided to perform static analysis by turning off the inertial terms for a certain time period to stabilize the system under the gravitational loading and pressure. The executioner and the post-processor blocks are provided to solve the nonlinear system and extract the required data.
```
Comment (hoffwm) [inline doc/content/examples/example6a.md]: Reword this and better explain why it is necessary.

**Comment by cbolisetti** [comment]
@hrishiv your tests are failing. Looks like some files are missing. Please check.

File: doc/content/examples/example6a.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+The mesh [ex06input] used in this model was generated externally using the mesh generation software, [Cubit](https://cubit.sandia.gov/). MASTODON imports the mesh shown in [fig:ex06_input] using the `FileMesh` type specified in the `Mesh` block. Details regarding supported mesh file types can be found at [GettingStarted](manuals/user/index.md).
+
+!media media/examples/ex06/ex06_input.png
+       style=width:50%;margin-left:150px;float:center;
+       id=fig:ex06_input
+       caption=Input model in MASTODON.
+
+The displacement variables are defined in the `Variables` block. The accelerations, velocities, normal and tangential (frictional) forces are defined as auxiliary variables in the `AuxVariables` block.
+
+[TensorMechanics](syntax/index.md) and [InertialForce](syntax/index.md) kernels are used to model the dynamics of the 3D solid elements. [Gravity](syntax/index.md) kernels is used to apply the gravitational force. The [Newmark time integration](manuals/theory/index.md) parameters used in this problem correspond to the Newmark's average acceleration method, i.e. `beta = 0.25` and `gamma = 0.5` and damping is ignored.
```
Comment (hoffwm) [inline doc/content/examples/example6a.md]: The Gravity block is specified in 'Kernels' in order to apply gravitational force to the entire model.

File: examples/ex06b/stabilize_isoil.i [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+    quantity = tangential_force_x
+    boundary = 102
+    paired_boundary = 103
+  [../]
+  [./tang_forc]
+    type = PenetrationAux
+    variable = tang_forc
+    quantity = tangential_force_magnitude
+    boundary = 102
+    paired_boundary = 103
```
Comment (hoffwm) [inline examples/ex06b/stabilize_isoil.i]: remove duplicate or unnecessary things here

File: examples/ex06c/stabilize_lc.i [hunk 10]
```diff
@@ -0,0 +1,352 @@
...
+  [./stress_soil]
+   type = ComputeFiniteStrainElasticStress
+   block = 1
+  [../]
+  [./den_soil]
+   type = GenericConstantMaterial
+   block = 1
+   prop_names = density
+   prop_values = 0.0001356 #slug/in^3
+#   prop_values = 0.3
```
Comment (hoffwm) [inline examples/ex06c/stabilize_lc.i]: remove

File: examples/ex06c/stabilize_lc.i [hunk 11]
```diff
@@ -0,0 +1,352 @@
...
+    block = 2
+  [../]
+  [./strain_block]
+    type = ComputeFiniteStrain
+    block = 2
+    displacements = 'disp_x disp_y disp_z'
+  [../]
+  [./stress_block]
+    type = ComputeFiniteStrainElasticStress
+  #  store_stress_old = true
```
Comment (hoffwm) [inline examples/ex06c/stabilize_lc.i]: remove

File: examples/ex06c/stabilize_lc.i [hunk 12]
```diff
@@ -0,0 +1,352 @@
...
+  [../]
+  [./strain_zx_interface]
+  type = SideAverageValue
+  boundary = 102
+  variable = strain_zx
+  [../]
+  [./stress_zx_interface]
+  type = SideAverageValue
+  boundary = 102
+  variable = stress_zx
```
Comment (hoffwm) [inline examples/ex06c/stabilize_lc.i]: Fix spacing here

File: examples/ex06c/stabilize_lc.i [hunk 13]
```diff
@@ -0,0 +1,352 @@
...
+
+
+
+
+
+
+[Outputs]
+  csv = true
+  exodus = true
+perf_graph  = true
```
Comment (hoffwm) [inline examples/ex06c/stabilize_lc.i]: tab

**Pull request merged**